tasks/amazon-6.yml

```diff
@@ -6,24 +6,31 @@
   register: probably_already_installed
 
 - block:
+
   - name: make temp dir for installation files
     file:
       path: /tmp/aws-cwa-inst
       state: directory
+
   - name: download & unpack the latest cloudwatch agent .zip
     unarchive:
       remote_src: yes
       src: "https://s3.amazonaws.com/amazoncloudwatch-agent/linux/amd64/latest/AmazonCloudWatchAgent.zip"
       dest: /tmp/aws-cwa-inst/
+
+  # Using documentation from AWS for installation:
+  # * https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/install-CloudWatch-Agent-on-first-instance.html#download-CloudWatch-Agent-on-EC2-Instance-first
   - name: install the RPM
-    yum:
-      name: /tmp/aws-cwa-inst/amazon-cloudwatch-agent.rpm
-      state: present
```

Comment:
I would guess that `state: latest` would implement the `-U` action, but I'll test that now, and if it doesn't work, go with yours.

Reply:
`state:latest` didn't work.
going with your changes.